src/Core/Components/CounterBadge/FluentCounterBadge.razor.cs

```diff
@@ -0,0 +1,77 @@
+// ------------------------------------------------------------------------
+// MIT License - Copyright (c) Microsoft Corporation. All rights reserved.
+// ------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Components;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// The FluentBadge component is a visual indicator that communicates a status or description of an associated component.
+/// It uses short text, color, and icons for quick recognition and is placed near the relavant content.
+/// </summary>
+public partial class FluentCounterBadge : FluentBadge, IFluentComponentBase
+{
+    private int _internalCount;
```

Comment:
Never create two variables containing the same thing: `_internalCount` and `Count`.
Later, you'll never know which variable to use.

It's better to have a method that returns what you need for the display in this case `<fluent-counter-badge count="@GetCount()" />`
```C#
private int? GetCount() => ShowWhen?.Invoke(Count) == true ? Count : null;
```